src/mempools/global_mempool.hpp

```diff
@@ -51,19 +51,21 @@ namespace argo {
 					auto nodes = backend::number_of_nodes();
 					memory = backend::global_base();
 					max_size = backend::global_size();
-					offset = new (&memory[0]) ptrdiff_t;
 					/**@todo this initialization should move to tools::init() land */
 					using namespace data_distribution;
 					base_distribution<0>::set_memory_space(nodes, memory, max_size);
+
+					// Move back one page as the last page is left for internal use
+					max_size -= reserved;
```

Comment:
Can this line not be merged with L53 to ensure that it is coherent with the `reset()` implementation?
The comments for both should maybe reflect what is "moved back".

Reply:
The line mentioned is moved further down due to:
https://github.com/etascale/argodsm/blob/756ebfac81eec4422811de1eaab1affeec07858e/src/mempools/global_mempool.hpp#L57